src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -98,18 +98,18 @@ public static object GetUninitializedObjectWithFormatterServices(Type type)
         private static HashSet<Type> registeredTypes;
         private static List<IExternalSerializer> externalSerializers;
         private static ConcurrentDictionary<Type, IExternalSerializer> typeToExternalSerializerDictionary;
-        private static Dictionary<string, Type> types;
-        private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
-        private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
-        private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
+        private static IReadOnlyDictionary<string, Type> types;
+        private static IReadOnlyDictionary<RuntimeTypeHandle, DeepCopier> copiers;
+        private static IReadOnlyDictionary<RuntimeTypeHandle, Tuple<Serializer,Deserializer>> serializers;
+        private static bool addingComplete;
         private static ConcurrentDictionary<Type, Func<GrainReference, GrainReference>> grainRefConstructorDictionary;
 
+
         private static IExternalSerializer fallbackSerializer;
         private static LoggerImpl logger;
         private static bool IsBuiltInSerializersRegistered;
         private static readonly object registerBuiltInSerializerLockObj = new object();
-        internal static int RegisteredTypesCount { get { return registeredTypes == null ? 0 : registeredTypes.Count; } }
```

Comment:
I can put it back, it wasn't threadsafe (as it was 2 calls to registeredTypes) and I couldn't find any references, but I can wrap it in a lock and throw it back in.